bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/BallerinaXmlSerializer.java

```diff
@@ -163,7 +195,12 @@ private void writeElement(XmlItem xmlValue) throws XMLStreamException {
         // Write attributes
         writeAttributes(currentNSLevel, attributeMap);
 
+        // Track and override xml text escape when xml text is within an element.
+        boolean prevWithinElementFlag = withinElement;
+        withinElement = true;
         xmlValue.getChildrenSeq().serialize(this);
+        withinElement = prevWithinElementFlag;
```

Comment:
Wouldn't it be clearer to use `this.withinElement` here?
```suggestion
        this.withinElement = true;
        xmlValue.getChildrenSeq().serialize(this);
        this.withinElement = prevWithinElementFlag;
```

Reply:
+1 updated.